hip/matrix/dense_kernels.hip.cpp

```diff
@@ -261,15 +579,49 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(
 template <typename ValueType>
 void transpose(std::shared_ptr<const HipExecutor> exec,
                matrix::Dense<ValueType> *trans,
-               const matrix::Dense<ValueType> *orig) GKO_NOT_IMPLEMENTED;
+               const matrix::Dense<ValueType> *orig)
+{
+    if (hipblas::is_supported<ValueType>::value) {
+        auto handle = exec->get_hipblas_handle();
+        {
+            hipblas::pointer_mode_guard pm_guard(handle);
+            auto alpha = one<ValueType>();
+            auto beta = zero<ValueType>();
+            hipblas::geam(handle, HIPBLAS_OP_T, HIPBLAS_OP_N,
+                          orig->get_size()[0], orig->get_size()[1], &alpha,
+                          orig->get_const_values(), orig->get_stride(), &beta,
+                          orig->get_const_values(), trans->get_size()[1],
+                          trans->get_values(), trans->get_stride());
+        }
+    } else {
+        GKO_NOT_IMPLEMENTED;
+    }
+};
 
 GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_TRANSPOSE_KERNEL);
 
 
 template <typename ValueType>
 void conj_transpose(std::shared_ptr<const HipExecutor> exec,
                     matrix::Dense<ValueType> *trans,
-                    const matrix::Dense<ValueType> *orig) GKO_NOT_IMPLEMENTED;
+                    const matrix::Dense<ValueType> *orig)
+{
+    if (hipblas::is_supported<ValueType>::value) {
+        auto handle = exec->get_hipblas_handle();
+        {
+            hipblas::pointer_mode_guard pm_guard(handle);
+            auto alpha = one<ValueType>();
+            auto beta = zero<ValueType>();
+            hipblas::geam(
+                handle, HIPBLAS_OP_C, HIPBLAS_OP_N, orig->get_size()[0],
+                orig->get_size()[1], &alpha, orig->get_const_values(),
+                orig->get_stride(), &beta, static_cast<ValueType *>(nullptr),
```

Comment:
Here we pass a `nullptr` to hipblas

Reply:
It should be the same pointer as `transpose`.
Hip does not provide the interface, so we do not face the test failed.
I will add it in next commit.